cmd/root.go

```diff
@@ -47,8 +48,11 @@ func AddCommands(cmd *cobra.Command) {
 		cfg.PublicMetrics.MetricsURL, "Push metrics to url")
 	cmd.PersistentFlags().DurationVar(&cfg.PublicMetrics.MetricsPushPeriod, "metrics-push-period",
 		cfg.PublicMetrics.MetricsPushPeriod, "Push period")
-	cmd.PersistentFlags().StringArrayVar(&cfg.PoETServers, "poet-server",
-		cfg.PoETServers, "The poet server url. (temporary) Can be passed multiple times")
```

Comment:
With `DeprecatedPoETServer` as an exported type we could instead define the flag as:

```go
cmd.PersistentFlags().AddFlag(&pflag.Flag{
	Name:     "poet-server",
	Value:    &cfg.PoETServers,
	DefValue: nil,
})
```

And add in `EnsureCLIFlags` a case for the `activation.DeprecatedPoETServer` type that logs the message to take a look at the release notes / readme and then panics.

Wdyt?

Reply:
This wouldn't work. Parsing of the flag would fail with a cryptic message complaining that it expected a map but got a slice before it even reached `EnsureCLIFlags`. Please check the updated code, I modified it to define the deprecation message in a single place and log it instead include in the error. I think the proposed solution is nice as it provides an easy way to add more deprecated fields & flags (which I'm sure will come in handy in the future).